autopush/web/registration.py

```diff
@@ -307,6 +337,15 @@ def _return_endpoint(self, endpoint_data, new_uaid, router=None):
                        client_info=self._client_info)
         self.finish()
 
+    def _list_channels(self, uaid):
+        _, channels = self.ap_settings.message.all_channels(str(uaid))
```

Comment:
this db call should definitely be in a thread though, we just don't really want the write/finish in one